thanos/playbooks/ThanosCompactHalted.md

```diff
@@ -0,0 +1,162 @@
+
+# Thanos Compactor Troubleshooting Playbook
+
+## Table of Contents
+
+1. [Thanos Compactor Halted Due to Overlapping Blocks](#thanos-compactor-halted-due-to-overlapping-blocks)
+2. [Thanos component has disappeared](#Thanos-component-has-disappeared)
+3. [Chunk critical error](#Chunk-critical-error)
+ 
+ ---
+
+## Thanos Compactor Halted Due to Overlapping Blocks
+
+### Problem
+The Thanos compactor has halted with an error similar to:
+```
+critical error detected; halting" err="compaction: ... pre compaction overlap check: overlaps found while gathering blocks. ...
+```
+Example:
+```
+s=2025-07-25T11:34:26.357181007Z caller=compact.go:559 level=error msg="critical error detected; halting" err="compaction: group 0@3105571489545500179: pre compaction overlap check: overlaps found while gathering blocks. [mint: 1742680710957, maxt: 1742680800000, range: 1m29s, blocks: 2]: <ulid: 01JQ01BAXKEBJSXT0QJ3PAQ63V, mint: 1742673600011, maxt: 1742680800000, range: 1h59m59s>, <ulid: 01JQ3P5BSQK656HMXMW7DREQ3D, mint: 1742680710957, maxt: 1742680800000, range: 1m29s>\n[mint: 1742680800517, maxt: 1742688000000, range: 1h59m59s, blocks: 2]: <ulid: 01JQ08725PXK4D6XD0SB1WYT2E, mint: 1742680800020, maxt: 1742688000000, range: 1h59m59s>, <ulid: 01JQ3P5EH930NGCFHTT9Z87TEY, mint: 1742680800517, maxt: 1742688000000, range: 1h59m59s>"
+```
+
+This is caused by overlapping blocks in your object storage.
+
+---
+
+### Resolution Steps
+
+#### 1. Identify Overlapping Blocks
+
+- Enter the compactor container or a pod with Thanos CLI access.
+- Run:
+  ```
+  thanos tools bucket inspect --objstore.config-file=<your-objstore.yaml> | grep <block_id>
+  ```
+- Look for blocks with overlapping `mint` and `maxt` time ranges. Note the ULIDs of the offending blocks.
+
+- Use `| grep <block_id>` to compare block sizes and timestamps from the error log.
+
+---
+
+#### 2. Decide Which Blocks to Delete
+
+- Prefer deleting the block with the **shorter time range** or the one that appears incomplete (smaller size). 
+- Example from logs:
+  - `01JQ3P5BSQK656HMXMW7DREQ3D`
+  - `01JQ3P5EH930NGCFHTT9Z87TEY`
```

Comment:
should we add the respective `du -sh` command for size comparison?

Reply:
not familiar with that command, i'll add output of thanos tools command. it shows the number of chunks of a block. this should be enough to compare between two overlapping ones

Reply:
ok